### ISSUE TITLE
Backend basepath issue

### ISSUE THREAD

**Issue by jschutte27** [Axway-API-Management-Plus/apim-cli#412]
### APIM-CLI version

1.140.0

### API-Management version

7.7.0.20230228

### Bug description

I am now testing 1.14.0 and seeing a difference from 1.13.4 which we have been using.  The backend basepath seems incorrect when importing a swagger file without a server.url value.

Here is the resulting output in api manager:
![image](https://github.com/Axway-API-Management-Plus/apim-cli/assets/58127265/1a7ccec1-da76-4dcd-931c-f2533b2e85b7)

![image](https://github.com/Axway-API-Management-Plus/apim-cli/assets/58127265/222d5806-f904-45b5-acf7-145e56839978)

Notice the resource path is "/".  This seems to be the main issue, but that slash appears to be added to the frontend as well.

Is this how the changes in 1.13.5+ are intended to work or it is a bug?  If it is intended, how should we be using it when servers.url is null?

### Steps to reproduce

Import command:
apim.sh api import -config "/opt/actions-runner/_work/it-axway-automation/it-axway-automation/testfiles/ordermgmt-api-config.json" --apidefinition "testfiles/ordermgmtswagger.json" -h *** -u *** -p '***' -force -forceUpdate -retryDelay 3000

api config:
```
{
  "name" : "Test Deploy Order Management Internal",
  "path" : "/ordermgmtdeploytesting",
  "state" : "published",
  "version" : "v1",
  "organization" : "IT Developers",
  "summary" : "Order Management Internal Only",
  "backendBasepath" : "https://server.domain.com/ordermanagement",
  "securityProfiles" : [ {
    "name" : "_default",
    "isDefault" : true,
    "devices" : [ {
      "name" : "API Key",
      "type" : "apiKey",
      "order" : 1,
      "properties" : {
        "apiKeyFieldName" : "KeyId",
        "takeFrom" : "HEADER",
        "removeCredentialsOnSuccess" : "true"
      }
    } ]
  } ]
}
```

### Relevant log output

```shell
2023-07-07 14:40:52,502 [APIManagerCLI] INFO : API-Manager CLI: 1.14.0
2023-07-07 14:40:52,504 [APIManagerCLI] INFO : Module: API - I M P O R T (1.14.0)
2023-07-07 14:40:52,511 [CoreParameters] INFO : Retrying unexpected API-Manager REST-API responses with a delay of 3000 milliseconds.
Jul 07, 2023 2:40:53 PM org.apache.http.client.protocol.ResponseProcessCookies processCookies
WARNING: Invalid cookie header: "Set-Cookie: spSessionId2=deleted;Version=1;Comment=;Domain=;Path=/api/portal/v1.4/;Max-Age=0;Secure;HttpOnly;SameSite=Strict". Blank or null value for domain attribute
2023-07-07 14:40:53,313 [APIManagerAdapter] INFO : Successfully connected to API-Manager (7.7.20230228) on: https://***:8075
2023-07-07 14:40:53,424 [APISpecificationFactory] INFO : Reading API-Definition (Swagger/WSDL) from file: testfiles/ordermgmtswagger.json (relative path)
2023-07-07 14:40:53,470 [APISpecificationFactory] INFO : Detected: Open API 3.0 specification. 
2023-07-07 14:40:53,471 [OAS3xSpecification] INFO : Updating openapi Servers url with value : https://server.domain.com/
2023-07-07 14:40:53,605 [APIImportApp] INFO : Lookup actual API based on Path: /ordermgmtdeploytesting  
2023-07-07 14:40:55,423 [APIManagerAPIAdapter] INFO : Loading application quotas for 1 subscribed applications.
2023-07-07 14:40:55,521 [APIImportManager] INFO : Re-Creating API as the ForceUpdate flag is set
2023-07-07 14:40:55,522 [RecreateToUpdateAPI] INFO : Create new API to update existing: Test Deploy Order Management Internal (ID: b2cca19e-6ec1-46cc-85d5-34321d5b16b4)
2023-07-07 14:40:55,761 [CreateNewAPI] INFO : Create published API: Test Deploy Order Management Internal v1  based on Open API 3.0 specification.
2023-07-07 14:40:55,869 [CreateNewAPI] INFO : Updating API backendBasePath with value : https://server.domain.com/ordermanagement
2023-07-07 14:40:55,974 [CreateNewAPI] INFO :  Successfully created published API: Test Deploy Order Management Internal v1 (ID: e3f865f8-24df-4673-9eec-0aa30fdc6668)
2023-07-07 14:40:55,975 [RecreateToUpdateAPI] INFO : New API successfully created. Going to delete old API: Test Deploy Order Management Internal v1 (ID: b2cca19e-6ec1-46cc-85d5-34321d5b16b4)
2023-07-07 14:40:56,153 [APIManagerAPIAdapter] INFO : API: Test Deploy Order Management Internal v1 ( b2cca19e-6ec1-46cc-85d5-34321d5b16b4 ) successfully deleted
```

**Comment by rathnapandi** [comment]
Hi @jschutte27, 

Use a parameter -overrideSpecBasePath on latest version. 
We did change the way it modifies the basepath, please refer #354

**Comment by jschutte27** [comment]
Hi @rathnapandi,

I tested this situation with the new -overrideSpecBasePath parameter and am running into an issue when servers.url is null.

The import log shows this line:
[OAS3xSpecification] INFO : Updating openapi Servers url with value : https://server.domain.com/

I would have expected it to be "https://server.domain.com/ordermanagement" from the config above.  

When servers.url is present the input log shows:
[OAS3xSpecification] INFO : overriding openapi Servers url with value : https://server.domain.com/ordermanagement

I feel like my use case with a null server.url should work without the overrideSpecBasePath parameter as it did before because it should just use the backend_basepath + servers.url.

Thanks,
Jeremiah

**Comment by rathnapandi** [comment]
Hi @jschutte27, 
The current code does not handle the scenario. I have fixed the issue, can you test with 
[axway-apimcli-1.14.1-SNAPSHOT.tar.gz](https://github.com/Axway-API-Management-Plus/apim-cli/files/12053471/axway-apimcli-1.14.1-SNAPSHOT.tar.gz) ?

Thanks
Rathna

**Comment by jschutte27** [comment]
This appears to fix the -overrideSpecBasePath parameter for me.  Thanks!

**Comment by rathnapandi** [comment]
Fixed in v1.14.1
